config/webpack.config.dev.js

```diff
@@ -179,6 +181,14 @@ module.exports = {
     // a plugin that prints an error when you attempt to do this.
     // See https://github.com/facebookincubator/create-react-app/issues/240
     new CaseSensitivePathsPlugin(),
+    // Warn on using .jsx files, prefer using .js only.
+    // See https://github.com/facebookincubator/create-react-app/issues/290
+    new CheckFilenamePlugin({
+      regex: /\.jsx$/,
```

Comment:
The reasoning behind choosing loader vs. plugin:

1. Loader can reuse path matching logic.
2. Loader functional scope is more restrictive: you see exactly what it affects vs. plugin which can do literally anything under the hood. I'd expect the less custom plugins the better, especially when you eject and start modifying config.